### PR TITLE
UIPQB-55: Regular expressions are incorrect for contains and starts_with operators

### PR DESCRIPTION
**Pull request by UladzislauKutarkin**
[UIPQB-55](https://issues.folio.org/browse/UIPQB-55)
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by zburke — APPROVED** [review]
It would be great to have some of the detail provided in the Jira ticket (which describes the bug) here in the PR title/description which should, instead of describing the problem, describe how the code is changing, e.g. "Avoid extra caret prefix in contains/starts_with queries" instead of "Regular expressions are incorrect...". 

A good way to think of PR titles is to complete the sentence, "After this is merged, the code will ..."

File: src/QueryBuilder/QueryBuilder/helpers/query.test.js [hunk 1]
```diff
@@ -122,8 +122,8 @@ describe('mongoQueryToSource()', () => {
         { user_last_name: { $lt: 10 } },
         { user_last_name: { $gte: 'value' } },
         { languages: { $in: ['value', 'value2'] } },
-        { user_full_name: { $regex: '^abc' } },
-        { user_full_name: { $regex: '^abc' } },
+        { user_full_name: { $regex: 'abc' } },
```
Comment (vashjs) [inline src/QueryBuilder/QueryBuilder/helpers/query.test.js:125]: Should this line be duplicated?

Comment (zburke) [reply src/QueryBuilder/QueryBuilder/helpers/query.test.js:125]: `initial` and `source` need to stay in sync. I think you're right that it's a duplicate, @vashjs, but we'd need to purge the related lines throughout the rest of this test file too: lines 62-67, and line 85.

**Review by vashjs — APPROVED** [review]

File: src/QueryBuilder/QueryBuilder/helpers/query.test.js (hunk 1 above)
Comment (vashjs) [reply src/QueryBuilder/QueryBuilder/helpers/query.test.js:125]: @zburke  thanks for response, you are right. I think we will do small tests clenup in feature stories.

**Review by vashjs — APPROVED** [review]

**Comment by vashjs** [comment]
> It would be great to have some of the detail provided in the Jira ticket (which describes the bug) here in the PR title/description which should, instead of describing the problem, describe how the code is changing, e.g. "Avoid extra caret prefix in contains/starts_with queries" instead of "Regular expressions are incorrect...".
> 
> A good way to think of PR titles is to complete the sentence, "After this is merged, the code will ..."

I completely agree, it’s also a good idea to add screenshots or GIFs where possible, for a better understanding of what has changed.

File: src/QueryBuilder/QueryBuilder/helpers/query.test.js (hunk 1 above)
Comment (UladzislauKutarkin) [reply src/QueryBuilder/QueryBuilder/helpers/query.test.js:118]: I removed duplicates. Thanks for your comments.

**Pull request merged**